src/mutations/collection/putCollection.ts

```diff
@@ -0,0 +1,83 @@
+import trim from 'lodash/trim'
+import { validate as validateUUID } from 'uuid'
+
+import { ASSET_TYPE } from 'common/enums'
+import {
+  ForbiddenError,
+  EntityNotFoundError,
+  UserInputError,
+  AssetNotFoundError,
+} from 'common/errors'
+import { fromGlobalId } from 'common/utils'
+import { MutationToPutCollectionResolver } from 'definitions'
+
+const resolver: MutationToPutCollectionResolver = async (
+  _,
+  { input: { id, title, description, cover } },
+  { dataSources: { collectionService, systemService }, viewer }
+) => {
+  if (!viewer.id) {
+    throw new ForbiddenError('Viewer has no permission')
+  }
+
+  if (title && title.length > 20) {
+    throw new UserInputError('Title too long')
+  }
+  if (description && description.length > 140) {
+    throw new UserInputError('Description too long')
+  }
+
+  if (cover && !validateUUID(cover)) {
+    throw new AssetNotFoundError('Asset does not exists')
+  }
+
+  if (id && fromGlobalId(id).type !== 'Collection') {
+    throw new UserInputError('Invalid Collection id')
+  }
+
+  let coverId
+  if (cover) {
+    const asset = await systemService.findAssetByUUID(cover)
+
+    if (
+      !asset ||
+      [ASSET_TYPE.embed, ASSET_TYPE.cover].indexOf(asset.type) < 0 ||
```

Comment:
`embed` & `cover` are for article cover, add a new asset type `collectionCover` for collection also?

Reply:
I don't get the reason to distinguish  article and collection cover

Reply:
Based on product/design, collection has a own optional cover, we should have another asset type for it like tag/user/circle?